src/main/java/com/minecolonies/coremod/datalistener/ResearchListener.java

```diff
@@ -92,7 +93,7 @@ protected void apply(@NotNull final Map<ResourceLocation, JsonElement> object, @
         // For dedicated servers, send to any connected players.  On startup, this will be no-one.
         // But it is possible to reload data packs live, and while not supported it's something to handle.
 
-        if(server instanceof DedicatedServer)
+        if(server instanceof DedicatedServer || (server instanceof IntegratedServer && server.getServerPort() > 0))
```

Comment:
Yeah, I think just doing a !world.isRemote check here should be just fine imo

Reply:
I'm not sure if there's a good world available.  The EventHandler case has event.getPlayer().world.isRemote(), but it's already wrapped to already check if the event.getPlayer() is a ServerPlayerEntity and returns false even if the player _is_ connecting from a separate machine.  For ResearchListener, doing something like server.getWorlds() seems like it's not correct, either.

In practice, I think these events only fire on the server side, so if there's no concerns about extraneous network messaging or performance given that they only happen once or twice a play session, can just simplify the checks.

Reply:
Do not check for sides unless you have to split calls from client and server because of data/logic, network is completely side safe

Reply:
Oh wait, this is on the server always, if you have the server object, you must be on the server, so a null check should be enough?

Reply:
Yeah, from a not-crashing perspective the null check's all you need for ResearchListener.  It's a bit of extra RAM and CPU for single-player instances, since they end up rebuilding the tree twice, but that's probably premature optimization.

I've got a few concerns about order-of-actions logic stuff, but the worst case is seeing autostart+instant research announcements triggering twice on server startup and even that may not be possible, so not worth the ugliness.